cmd/ui/src/views/Explore/ContextMenu/AssetGroupMenuItem.tsx

```diff
@@ -0,0 +1,158 @@
+
+    const [open, setOpen] = useState(false);
+
+    const selectedNode = useSelector((state: AppState) => state.entityinfo.selectedNode);
+    const tierZeroAssetGroupId = useSelector(selectTierZeroAssetGroupId);
+
+    const isMenuItemForTierZero = assetGroupId === tierZeroAssetGroupId;
+
+    const mutation = useMutation({
+        mutationFn: ({ nodeId, action }: { nodeId: string; action: 'add' | 'remove' }) => {
+            return apiClient.updateAssetGroupSelector(assetGroupId, [
+                {
+                    selector_name: nodeId,
+                    sid: nodeId,
+                    action,
+                },
+            ]);
+        },
+        onSuccess: () => {
+            if (selectedNode?.graphId && isMenuItemForTierZero) {
+                dispatch(toggleTierZeroNode(selectedNode.graphId));
+            }
+
+            addNotification(
+                'Update successful. Please check back later to view updated Asset Group.',
+                'AssetGroupUpdateSuccess'
+            );
+        },
+        onError: (error: any) => {
+            console.error(error);
+            addNotification('Unknown error, group was not updated', 'AssetGroupUpdateError');
+        },
+    });
+
+    const { data: assetGroupMembers } = useQuery(['listAssetGroupMembers', assetGroupId], () =>
+        apiClient
+            .listAssetGroupMembers(assetGroupId, undefined, {
+                params: {
+                    object_id: `object_id=eq:${selectedNode?.id}`,
+                },
+            })
+            .then((res) => res.data.data?.members)
+    );
+
+    const handleAddToAssetGroup = () => {
+        if (selectedNode) {
+            mutation.mutate({ nodeId: selectedNode.id, action: 'add' });
+        }
+    };
+
+    const handleRemoveFromAssetGroup = () => {
+        if (selectedNode) {
+            mutation.mutate({ nodeId: selectedNode.id, action: 'remove' });
+        }
+    };
+
+    const handleOpenConfirmation = (e: React.MouseEvent<HTMLLIElement>) => {
+        e.stopPropagation();
+        setOpen(true);
+    };
+
+    const handleCloseConfirmation = () => {
+        setOpen(false);
+    };
+
+    // error state, data didn't load
+    if (!assetGroupMembers) {
+        return null;
+    }
+
+    // selected node is not a member of the group
+    if (assetGroupMembers.length === 0) {
+        return (
+            <>
+                <MenuItem onClick={isMenuItemForTierZero ? handleOpenConfirmation : handleAddToAssetGroup}>
+                    Add to {assetGroupName}
+                </MenuItem>
+                {isMenuItemForTierZero ? (
+                    <ConfirmationDialog
+                        handleCancel={handleCloseConfirmation}
+                        handleApply={handleAddToAssetGroup}
+                        open={open}
+                        dialogContent={`Are you sure you want to add this node to ${assetGroupName}? This action will initiate an analysis run to update group membership.`}
```

Comment:
I think we might want to change the confirmation messages here, since updating membership no longer requires waiting on an analysis run

Reply:
we should leave this message, because this message is explicitly when adding/removing a node to the high value asset group.